Client/Coordinators/BaseCoordinator.swift

```diff
@@ -5,7 +5,7 @@
 import Common
 import Foundation
 
-open class BaseCoordinator: Coordinator {
+open class BaseCoordinator: NSObject, Coordinator {
```

Comment:
Why was `NSObject` needed?

Reply:
`NSObject` needed because of this error:

`/firefox-ios/Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift:68:1: Cannot declare conformance to 'NSObjectProtocol' in Swift; 'EnhancedTrackingProtectionCoordinator' should inherit 'NSObject' instead`

Reply:
Ah ok when we conformed to `UIViewControllerTransitioningDelegate` 👍